src/ext/java/org/opentripplanner/ext/reportapi/resource/ReportResource.java

```diff
@@ -62,4 +70,26 @@ public Response getBicycleSafetyAsCsv(
       )
       .build();
   }
+
+  @GET
+  @Path("/graph.json")
+  public synchronized Response stats(@Context OtpServerRequestContext serverRequestContext) {
+    // since the computation is pretty expensive only allow it every 5 minutes
+    if (cachedStats == null || cachedStats.hasExpired(Instant.now())) {
+      var stats = GraphReportBuilder.build(serverRequestContext);
+      cachedStats = new CachedStats(Instant.now(), stats);
+    }
+
+    return Response
+      .status(Response.Status.OK)
+      .entity(cachedStats.stats())
+      .type("application/json")
+      .build();
+  }
+
+  private record CachedStats(Instant creationTime, GraphReportBuilder.GraphStats stats) {
+    public boolean hasExpired(Instant time) {
+      return this.creationTime.plus(Duration.ofMinutes(5)).isBefore(time);
+    }
+  }
```

Comment:
This do not implement the cashing as indicated by the name, you could move the hasExpired inside and then ask for the latest `stats ` with something like this:

```Java
static class Cache<T> {
    private T value;
    private long timeout;
    private final Duration cacheInterval;

    public Cache(T value, Duration cacheInterval) {
      this.value = Objects.requireNonNull(value);
      this.cacheInterval = cacheInterval;
      this.timeout = calculateTimeout();
    }

    private long calculateTimeout() {
      return Instant.now().plus(cacheInterval).toEpochMilli();
    }

    public T get(Supplier<T> refetchValue) {
      synchronized (this) {
        if(hasExpired()) {
          this.value = refetchValue.get();
          this.timeout = calculateTimeout();
        }
      }
      return value;
    }
    private boolean hasExpired() {
      return timeout < System.currentTimeMillis();
    }
  }

```

Note! this synchronized on it self, whitch is ok since that is its only purpose - it should be documented in the class JavaDoc. This a generic implementation, so we can put it in the utils somewhere.

Reply:
I used the `System.currentTimeMillis()` in the `hasExpiered()`, but you can use `Instant` if you want, it comes down to taste/pros&cons.